File: ui/src/main/kotlin/listenwithrandos/app/Navbar.kt
package listenwithrandos.app

import react.RBuilder
import react.dom.div
import react.dom.li
import react.dom.nav
import react.dom.ul
import react.router.dom.routeLink

fun RBuilder.navbar() {
    nav(classes = "navbar") {
        div(classes = "container") {
            routeLink(to = "/", className = "navbar-brand") {
                +"Listen With Randos"
            }

            ul(classes = "navbar-nav") {
                connectedLoginSwitch {
                    attrs {
                        loggedInRender = {
                            li(classes = "nav-item active") {
                                connectedLogoutButton {}
                            }
                        }
                        loggedOutRender = {
                            li(classes = "nav-item active") {
                                loginLink(classes = "nav-link")
                            }
                        }
                    }
                }
            }
        }
    }
}

File: ui-new/src/app/Navbar/Navbar.js
// @flow

import * as React from 'react'
import {Link} from 'react-router-dom'
import {LoggedInRender, LoggedOutRender} from '../LoginSwitch/LoginSwitch'
import {LoginLink} from '../LoginLink/LoginLink'
import {LogoutButton} from './LogoutButton/LogoutButton'

export const Navbar = () => (
  <nav className="navbar">
    <div className="container">
      <Link to="/" className="navbar-brand">Listen With Randos</Link>

      <ul className="navbar-nav">
        <LoggedInRender>
          <li className="nav-item">
            <LogoutButton/>
          </li>
        </LoggedInRender>

        <LoggedOutRender>
          <li className="nav-item">
            <LoginLink>Log in</LoginLink>
          </li>
        </LoggedOutRender>
      </ul>
    </div>
  </nav>
)
